--- gzip-1.9/gzexe.in.orig	2018-06-12 20:56:23.836235854 +0200
+++ gzip-1.9/gzexe.in	2018-06-12 21:03:07.269564583 +0200
@@ -169,7 +169,7 @@
 if command -v mktemp >/dev/null 2>&1; then
   gztmpdir=`mktemp -d "${TMPDIR}gztmpXXXXXXXXX"`
 else
-  gztmpdir=${TMPDIR}gztmp$$; mkdir $gztmpdir
+  echo 'cannot create temporary file: missing mktemp program' >&2; exit 1;
 fi || { (exit 127); exit 127; }
 
 gztmp=$gztmpdir/$0
