--- gzip-1.3.6/zdiff.in.orig	2006-11-29 23:57:02.078114250 +0100
+++ gzip-1.3.6/zdiff.in	2006-11-30 00:06:39.394194250 +0100
@@ -60,11 +60,12 @@
 	        *[-.]gz* | *[-.][zZ] | *.t[ga]z)
 			F=`echo "$2" | sed 's|.*/||;s|[-.][zZtga]*||'`
 			set -C
-			trap 'rm -f /tmp/"$F".$$; exit 2' HUP INT PIPE TERM 0
-			gzip -cdfq -- "$2" > /tmp/"$F".$$ || exit
-			gzip -cdfq -- "$1" | $comp $OPTIONS - /tmp/"$F".$$
+			tmpfile=`mktemp ${TMPDIR:-/tmp}/gz.XXXXXX` || exit 1
+			trap 'rm -f $tmpfile; exit 2' HUP INT PIPE TERM 0
+			gzip -cdfq -- "$2" >| $tmpfile || exit
+			gzip -cdfq -- "$1" | $comp $OPTIONS - $tmpfile
                         STAT="$?"
-			/bin/rm -f /tmp/"$F".$$ || STAT=2
+			/bin/rm -f $tmpfile || STAT=2
 			trap - HUP INT PIPE TERM 0
 			exit $STAT;;
 
--- gzip-1.3.6/znew.in.orig	2006-11-29 23:57:02.082114500 +0100
+++ gzip-1.3.6/znew.in	2006-11-30 00:09:00.795031250 +0100
@@ -56,9 +56,9 @@
 # block is the disk block size (best guess, need not be exact)
 
 warn="(does not preserve modes and timestamp)"
-tmp=/tmp/zfoo.$$
+tmp=`mktemp ${TMPDIR:-/tmp}/zfoo.XXXXXX` || exit 1
 set -C
-echo hi > $tmp || exit
+echo hi >| $tmp || exit
 if test -z "`(${CPMOD-cpmod} $tmp $tmp) 2>&1`"; then
   cpmod=${CPMOD-cpmod}
   warn=""
@@ -114,6 +114,7 @@
 
 for i do
   n=`echo $i | sed 's/.Z$//'`
+  ntmp=`mktemp ${TMPDIR:-/tmp}/$n.XXXXXX` || exit 1
   if test ! -f "$n.Z" ; then
     echo $n.Z not found
     res=1; continue
@@ -129,7 +130,7 @@
     fi
   else
     if test $check -eq 1; then
-      if cp -p "$n.Z" "$n.$$" 2> /dev/null || cp "$n.Z" "$n.$$"; then
+      if cp -p "$n.Z" $ntmp 2> /dev/null || cp "$n.Z" $ntmp; then
 	:
       else
 	echo cannot backup "$n.Z"
@@ -139,7 +140,7 @@
     if gzip -d "$n.Z"; then
       :
     else
-      test $check -eq 1 && mv "$n.$$" "$n.Z"
+      test $check -eq 1 && mv $ntmp "$n.Z"
       echo error while uncompressing $n.Z
       res=1; continue
     fi
@@ -147,7 +148,7 @@
       :
     else
       if test $check -eq 1; then
-	mv "$n.$$" "$n.Z" && rm -f "$n"
+	mv $ntmp "$n.Z" && rm -f "$n"
         echo error while recompressing $n
       else
 	# compress $n  (might be dangerous if disk full)
@@ -162,7 +163,7 @@
     if test $pipe -eq 1; then
       rm -f "$n$ext"
     elif test $check -eq 1; then
-      mv "$n.$$" "$n.Z" && rm -f "$n$ext"
+      mv $ntmp "$n.Z" && rm -f "$n$ext"
     else
       gzip -d "$n$ext" && compress "$n" && rm -f "$n$ext"
     fi
@@ -170,9 +171,9 @@
 
   elif test $check -eq 1; then
     if gzip -t "$n$ext" ; then
-      rm -f "$n.$$" "$n.Z"
+      rm -f $ntmp "$n.Z"
     else
-      test $pipe -eq 0 && mv "$n.$$" "$n.Z"
+      test $pipe -eq 0 && mv $ntmp "$n.Z"
       rm -f "$n$ext"
       echo error while testing $n$ext, $n.Z unchanged
       res=1; continue
