--- gzip-1.3.6/gzexe.in.orig	2006-11-20 09:40:33.000000000 +0100
+++ gzip-1.3.6/gzexe.in	2006-11-29 23:50:44.730531500 +0100
@@ -132,12 +132,12 @@
     (sed 1q $0 &&
      sed "s|^if tail|if $tail|" <<'EOF' &&
 skip=26
-set -C
+
 umask=`umask`
 umask 77
-if (tempfile --version) >/dev/null 2>&1
-then gztmp=`tempfile -p gztmp` || exit
-else gztmp=/tmp/gztmp$$
+if (mktemp -V) >/dev/null 2>&1
+then gztmp=`mktemp ${TMPDIR:-/tmp}/gztmp$$.XXXXXX` || exit
+else { echo 'cannot create temporary file' >&2; exit 1; }
 fi
 if tail +$skip "$0" | "BINDIR"/gzip -cd > "$gztmp"; then
   umask $umask
