--- gzip-1.3/gzip.c~	Thu Nov 11 18:57:25 1999
+++ gzip-1.3/gzip.c	Wed Dec 12 16:33:42 2001
@@ -317,19 +317,18 @@
 /* ======================================================================== */
 local void usage()
 {
-    printf ("usage: %s [-%scdfhlLnN%stvV19] [-S suffix] [file ...]\n",
-	    progname,
 #if O_BINARY
-	    "a",
+#define BINARY_FLAG "a"
 #else
-	    "",
+#define BINARY_FLAG ""
 #endif
 #ifdef NO_DIR
-	    ""
+#define DIR_FLAG ""
 #else
-	    "r"
+#define DIR_FLAG "r"
 #endif
-	    );
+    printf ("usage: %s [-%scdfhlLnN%stvV19] [-S suffix] [file ...]\n",
+	    progname, BINARY_FLAG, DIR_FLAG);
 }
 
 /* ======================================================================== */
