--- gzip-1.3.12/m4/extensions.m4~	2006-10-12 10:01:18.000000000 +0200
+++ gzip-1.3.12/m4/extensions.m4	2008-08-07 00:13:58.996890489 +0200
@@ -16,6 +16,7 @@
 # ------------------------
 # Enable extensions on systems that normally disable them,
 # typically due to standards-conformance issues.
+m4_ifndef([AC_USE_SYSTEM_EXTENSIONS],[
 AC_DEFUN([AC_USE_SYSTEM_EXTENSIONS],
 [
   AC_BEFORE([$0], [AC_COMPILE_IFELSE])
@@ -49,6 +50,7 @@
   AC_DEFINE([_POSIX_PTHREAD_SEMANTICS])
   AC_DEFINE([_TANDEM_SOURCE])
 ])
+])
 
 # gl_USE_SYSTEM_EXTENSIONS
 # ------------------------
